cmd/capibmadm/cmd/powervs/network/network.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+)
+
+// NewNetworkCommand function to add PowerVS network commands.
+func NewNetworkCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "network",
+		Short: "Perform PowerVS network operations",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			// we cannot have multiple PersistentPreRunE for a command
+			// This block is a workaround mentioned in https://github.com/spf13/cobra/issues/252
+			root := cmd
+			for ; root.HasParent(); root = root.Parent() {
+			}
+			if err := root.PersistentPreRunE(cmd, args); err != nil {
+				return err
+			}
+			if options.ServiceInstanceID == "" {
+				return fmt.Errorf("service-instance-id is not provided")
+			}
+			return nil
+		},
+	}
+	cmd.PersistentFlags().StringVar(&options.ServiceInstanceID, "service-instance-id", "", "PowerVS service instance id")
```

Comment:
this may end up registering in most of the subcommands under powervs(except keys), so wondering if we can create some utility function to registry the common flags to any given command